core/src/banking_stage.rs

```diff
@@ -1397,21 +1399,28 @@ impl BankingStage {
                 gossip_vote_sender,
             );
 
+        let mut unlock_time = Measure::start("unlock_time");
+        // Once the accounts are new transactions can enter the pipeline to process them
+        drop(batch);
+        unlock_time.stop();
+
         let ExecuteAndCommitTransactionsOutput {
             ref mut retryable_transaction_indexes,
             ref execute_and_commit_timings,
             ..
         } = execute_and_commit_transactions_output;
 
+        QosService::update_or_remove_transaction_costs(
+            transaction_costs.iter(),
+            transactions_qos_results.iter(),
+            retryable_transaction_indexes,
```

Comment:
@jstarry 's [comment](https://github.com/solana-labs/solana/pull/24182#discussion_r846204232) should be address. I'd suggested create issue to work on separately, as that is not as pressing as those retryable failures. wdyt?

Reply:
added issue #24197

Reply:
I agree. The current state is already an improvement, even if a part of the problem still remains.